iocore/net/SSLSNIConfig.cc

```diff
@@ -108,8 +116,18 @@ actionVector *
 SNIConfigParams::get(cchar *servername) const
 {
   auto actionVec = sni_action_map.get(servername);
-  if (!actionVec)
-    actionVec = wild_sni_action_map.get(servername);
+  if (!actionVec) {
+    Vec<cchar *> keys;
+    wild_sni_action_map.get_keys(keys);
+    for (int i = 0; i < static_cast<int>(keys.length()); i++) {
+      constexpr auto npos = ts::string_view::npos;
+      ts::string_view sv{servername, strlen(servername)};
+      auto *keyStr = keys.get(i);
+      if (sv.find(keyStr) != npos) {
```

Comment:
This will find any string in the servername, but it might not be a wildcard match.  For example if you have the wildcard *.foo.com it will be in the servername foo.com.net.

Reply:
you are right, I want `rfind` here not `find`

Reply:
What about
```
if (sv.size() >= keyStr->size() && sv.substr(sv.size() - keyStr->size()) == keyStr) {
  // it's a match
```